components/Tabs/src/private/MDCItemBar.m

```diff
@@ -170,26 +170,32 @@ - (void)setItems:(NSArray<UITabBarItem *> *)items {
     [self stopObservingItems];
 
     _items = [items copy];
-    // Update the path to the selected item.
-    if (_lastSelectedIndexPath) {
-      if (_selectedItem) {
-        NSUInteger index = [_items indexOfObject:_selectedItem];
-        if (NSNotFound != index) {
-          _lastSelectedIndexPath = [NSIndexPath indexPathForItem:index inSection:0];
-        } else {
-          _lastSelectedIndexPath = nil;
-        }
+
+    // Determine new selected item.
+    UITabBarItem *newSelectedItem = nil;
```

Comment:
It is redundant under ARC, but I would argue that being explicit with default values improves readability by removing a mental step for readers. Regardless, there was a better way to write this that avoids the issue.